swagger.yaml

```diff
@@ -1,78 +1,356 @@
-          schema:
-            $ref: '#/definitions/Info'
-  /sample:
-    get:
-      summary: Get message
-      description: This is a sample resource that returns a short message.
+      tags:
+        - directory
+      operationId: getDirectoryEntries
+      description: "Get all entities in directory. Can be filtered by query parameters"
+      produces:
+        - application/json
+      parameters:
+        - name: Full Name
+          in: query
+          description: "Full name of person to be searched for."
+          required: false
+          type: string
+
+        - name: Surname
+          in: query
+          description: "Surname of person to be searched for."
+          required: false
+          type: string
+
+        - name: Email Address
+          in: query
+          description: "Email address of person to be searched for."
+          required: false
+          type: string
+          format: email
+
+        - name: Office Phone
+          in: query
+          description: "Office phone number of person to be searched for."
+          required: false
+          type: string
+
+        - name: Alternate Phone
+          in: query
+          description: "Alternate phone number of person to be searched for."
+          required: false
+          type: string
+
+        - name: Home Phone
+          in: query
+          description: "Home phone number of person to be searched for."
+          required: false
+          type: string
+
+        - name: Fax Number
+          in: query
+          description: "Fax number of person to be searched for."
+          required: false
+          type: string
+
+        - name: Office Address
+          in: query
+          description: "Office address of person to be searched for."
+          required: false
+          type: string
+
+        - name: Department
+          in: query
+          description: "Department of person to be searched for."
+          required: false
+          type: string
+
+        - name: Affiliation
+          in: query
+          description: "Any, Employee, Student, Other are valid options to be searched with."
+          required: false
+          type: string
+
+        - name: Any Phone Number
+          in: query
+          description: "Any phone number to be searched with"
+          required: false
+          type: string
+
+        - name: Join Search Parameters With And or Or
+          in: query
+          description: "and & or are valid options. And is default value"
+          required: false
+          type: string
+
       responses:
-        200:
-          description: A short message
+        "200":
+          description: "Successful response"
           schema:
-            type: string
-        404:
-          description: Not found
+            $ref: "#/definitions/directoryEntity"
+        "400":
+          description: "Bad Request"
+          #example: "Error: Too many entries returned. Try a more precise search."
```

Comment:
the error response will generally be a json error object.

we will have to think about how to handle broad searches. for example, do we return the entire directory if someone supplies no parameters? do we implement paging?